hoverkite-protocol/src/response.rs

```diff
@@ -0,0 +1,189 @@
+use crate::Side;
+use std::collections::VecDeque;
+use std::convert::TryInto;
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Response {
+    pub side: Side,
+    pub response: SideResponse,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum SideResponse {
+    Log(String),
+    Position(i64),
+    BatteryReadings {
+        battery_voltage: u16,
+        backup_battery_voltage: u16,
+        motor_current: u16,
+    },
+    ChargeState {
+        charger_connected: bool,
+    },
+}
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct UnexpectedResponse(pub u8);
+
+impl Response {
+    pub fn parse(
+        buffer: &mut VecDeque<u8>,
+        side: Side,
+    ) -> Result<Option<Response>, UnexpectedResponse> {
+        match buffer.front().copied() {
+            Some(b'"') | Some(b'\'') => {
```

Comment:
(fine like this for now) I suppose by using " and ' rather than wrapping the log line in F, we end up with a SideResponse enum that isn't serialisable, and we're stuck adding two parsing code paths for every variant that we add to the enum.

Relatedly, I don't think that we need the length bye on the F message anymore, because we now parse the underlying command and re-serialise it, so we could add F here with or without a length bye and it would be fine.

Reply:
I agree that we should use F. I'll change it in a later PR.